packages/block-library/src/heading/edit.js

```diff
@@ -13,48 +13,30 @@ import HeadingToolbar from './heading-toolbar';
  */
 import { __ } from '@wordpress/i18n';
 import { PanelBody } from '@wordpress/components';
-import { compose } from '@wordpress/compose';
 import { createBlock } from '@wordpress/blocks';
 import {
 	AlignmentToolbar,
 	BlockControls,
 	InspectorControls,
 	RichText,
-	withColors,
-	PanelColorSettings,
+	__experimentalUseColors,
 } from '@wordpress/block-editor';
-import { memo } from '@wordpress/element';
-
-const HeadingColorUI = memo(
-	function( {
-		textColorValue,
-		setTextColor,
-	} ) {
-		return (
-			<PanelColorSettings
-				title={ __( 'Color Settings' ) }
-				initialOpen={ false }
-				colorSettings={ [
-					{
-						value: textColorValue,
-						onChange: setTextColor,
-						label: __( 'Text Color' ),
-					},
-				] }
-			/>
-		);
-	}
-);
 
 function HeadingEdit( {
 	attributes,
 	setAttributes,
 	mergeBlocks,
 	onReplace,
 	className,
-	textColor,
-	setTextColor,
 } ) {
+	const { TextColor, InspectorControlsColorPanel } = __experimentalUseColors(
+		[ { name: 'textColor', attribute: 'color' } ],
```

Comment:
At the start, our colors API explicitly made block developers identify what was the attribute and what was the CSS property:
https://github.com/WordPress/gutenberg/blob/b8ee131d764aa842a859a1c52fc847a4115fd1e1/core-blocks/button/index.js#L93-L94

People provided feedback in reviews that maybe we could just use something in this format { textColor: 'color' }, where the key represents the Gutenberg attribute and the value of the CSS context/property where the color is used. Would not this format fit here? What were the reasons for the change and go back to explicitly identify the Gutenberg attribute and CSS property?

We have functions and docs where we call the CSS property 'color' the color context. And currently, some code that exists outside of the hook calls 'textColor' colorAttributeName. I think naming 'color' attribute may be confusing. Could we keep the current name 'context'?

Reply:
`{ textColor: 'color' }` is super confusing and not self explanatory.

`color` here is an attribute, it's not context. I think it makes a lot more sense to stick to the terminology that the web platform uses.

Reply:
Hi @epiqueras, The problem is that the term attribute in Gutenberg has a very specific meaning. I think people will get confused.
On the web, we normally call "color", "background-color" CSS properties. I think if we call "property" we still stick to the terminology that the web platform uses while removing a source of possible confusion for developers.

Reply:
You're right, `property` makes more sense.